wasm/wat/parser_test.go

```diff
@@ -11,20 +11,25 @@ import (
 // example has a work-in-progress of supported functionality, used primarily for benchmarking. This includes:
 // * module and function names
 // * explicit, and inlined type definitions (including anonymous)
+// * inlined and overridden type parameter names
 // * start function
 //
-// NOTE: this is different from exampleWat because the parser doesn't yet support all features
+// Note: This is different from exampleWat because the parser doesn't yet support all features.
+// Note: When changing this, regenerate example.wasm like `wat2wasm --debug-names --debug-parser -v example.wat`
+// TODO: this has drifted because neither wabt, nor wasm-tools properly encodes the name section.
```

Comment:
so basically `names.go` hasn't been updated and neither has `testdata/example.wasm` I'll do that in the next PR